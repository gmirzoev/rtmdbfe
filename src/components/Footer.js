import React from 'react';
import * as styles from './Footer.scss';

const Footer = () => (
  <footer className={styles.footer}>
    <span className={styles.poweredByText}>
      This product uses the TMDb API but is not endorsed or certified by TMDb.
    </span>
    <div className={styles.poweredByLogo} />
  </footer>
);

export default Footer;
