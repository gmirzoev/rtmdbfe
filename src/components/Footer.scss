.footer {
  background-color: #0d7575;
  padding: 20px;
}

.powered-by {
  &-text {
    line-height: 60px;
    color: #fff;
    font-weight: bold;
  }

  &-logo {
    background-image: url('assets/images/powered-by.svg');
    width: 152px;
    height: 60px;
    background-size: cover;
    float: right;
  }
}
