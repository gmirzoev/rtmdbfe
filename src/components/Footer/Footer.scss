.footer {
  display: flex;
  flex-direction: row;
  align-items: center;
  background-color: var(--bg-color-main-dark);
  padding: 10px 20px;
}

.powered-by {
  &-text {
    color: var(--color-white);
    font-weight: bold;
    flex: 1;
    margin-right: 10px;
  }

  &-logo {
    background-image: url('assets/images/powered-by.svg');
    width: 130px;
    height: 50px;
    background-size: cover;
  }
}

@media (min-width: 481px) {
  .powered-by {
    &-logo {
      float: right;
    }
  }
}
