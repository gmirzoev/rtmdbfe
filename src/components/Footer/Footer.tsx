import * as React from 'react'
const styles = require('./Footer.scss')

export default class Footer extends React.Component {
  render() {
    return (
      <footer className={styles.footer}>
        <span className={styles.poweredByText}>
          This product uses the TMDb API but is not endorsed or certified by TMDb.
        </span>
        <div className={styles.poweredByLogo} />
      </footer>
    )
  }
}
