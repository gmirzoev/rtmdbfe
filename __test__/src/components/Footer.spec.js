import React from 'react';
import { shallow } from 'enzyme';
import Footer from 'components/Footer';

describe('Footer component', () => {
  it('should render correctly', () => {
    const component = shallow(<Footer />);
    expect(component).toMatchSnapshot();
  });
});
