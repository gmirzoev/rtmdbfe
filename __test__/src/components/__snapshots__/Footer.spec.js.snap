// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component should render correctly 1`] = `
<footer>
  <span>
    This product uses the TMDb API but is not endorsed or certified by TMDb.
  </span>
  <div />
</footer>
`;
